File: js/1956.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [V, E] = inputs[0].split(' ').map(Number);

const arr = Array(V + 1)
  .fill()
  .map(() => Array(V + 1).fill(Infinity));

for (let i = 1; i <= E; i++) {
  const [a, b, c] = inputs[i].split(' ').map(Number);
  arr[a][b] = c;
}

for (let k = 1; k <= V; k++) {
  for (let i = 1; i <= V; i++) {
    for (let j = 1; j <= V; j++) {
      arr[i][j] = Math.min(arr[i][j], arr[i][k] + arr[k][j]);
    }
  }
}
a
let answer = Infinity;
for (let i = 1; i <= V; i++) {
  answer = Math.min(answer, arr[i][i]);
}

console.log(answer === Infinity ? -1 : answer);

File: c++/1956.cpp
#include <iostream>

#define INF 1e9

using namespace std;

int arr[401][401];

int main() {
  freopen("example.txt", "r", stdin);
  ios_base::sync_with_stdio(0);
  cin.tie(0);

  int V, E;
  cin >> V >> E;

  for (int i = 1; i <= V; i++) {
    for (int j = 1; j <= V; j++) {
      arr[i][j] = INF;
    }
  }

  while (E--) {
    int a, b, c;
    cin >> a >> b >> c;
    arr[a][b] = c;
  }

  for (int k = 1; k <= V; k++) {
    for (int i = 1; i <= V; i++) {
      for (int j = 1; j <= V; j++) {
        arr[i][j] = min(arr[i][j], arr[i][k] + arr[k][j]);
      }
    }
  }

  int answer = INF;
  for (int i = 1; i <= V; i++) {
    if (arr[i][i] < answer) answer = arr[i][i];
  }

  cout << ((answer == INF) ? -1 : answer);

  return 0;
}